### PR TITLE
fix(dedicated.vps): add tracking for upscale and upgrade

### PR DESCRIPTION
**Pull request by stif59100**
ref:MANAGER-10312

Signed-off-by: stif59100 <steeve.vanderstocken@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `release/sprint-2250-2252` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-10312
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- ~~[ ] Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

## Description

I add tracking for upscale and upgrade for the vps on manager dedicated.

### CODE REVIEW

**Review by sachinrameshn — DISMISSED** [review]

File: packages/manager/modules/vps/src/upscale/upscale.component.js [hunk 1]
```diff
@@ -13,8 +13,9 @@ export default {
     scrollToTop: '<',
     upscaleOptions: '<',
     vps: '<stateVps',
-
     goBack: '<',
+    trackPage: '<',
+    trackingPrefix: '<',
```
Comment (mohammed-zahaf) [inline packages/manager/modules/vps/src/upscale/upscale.component.js:18]: Seems not required, to removed

File: packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js [hunk 2]
```diff
@@ -16,6 +16,9 @@ export default /* @ngInject */ ($stateProvider) => {
       },
       'configurationTile@vps.detail.dashboard': 'vpsDashboardTileConfiguration',
     },
+    params: {
+      trackingTag: null,
```
Comment (mohammed-zahaf) [inline packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js:20]: Not required i guess

**Review by sachinrameshn — DISMISSED** [review]

**Review by mohammed-zahaf — DISMISSED** [review]

File: packages/manager/modules/vps/src/upscale/upscale.controller.js [hunk 3]
```diff
@@ -479,8 +489,10 @@ export default class UpscaleController {
 
     this.loading.performUpscale = true;
     const planCode = this.planCode || this.range.planCode;
+    const currentPlanCode = this.upscaleRanges.find((e) => e.isCurrentRange)
+      .planCode;
```
Comment (anooparveti) [inline packages/manager/modules/vps/src/upscale/upscale.controller.js:493]:
```suggestion
    const currentPlanCode = this.upscaleRanges.find((e) => e.isCurrentRange)?.planCode;
```

File: packages/manager/modules/vps/src/upscale/upscale.controller.js [hunk 4]
```diff
@@ -479,8 +489,10 @@ export default class UpscaleController {
...
 
-    return this.performUpscale(planCode)
+    return this.performUpscale(planCode, currentPlanCode)
```
Comment (anooparveti) [inline packages/manager/modules/vps/src/upscale/upscale.controller.js:495]: **question:**  Why is this change made. I dont see its usage.

Comment (stif59100) [reply packages/manager/modules/vps/src/upscale/upscale.controller.js:495]: it's use on

.then(() => {
        this.atInternet.trackPage({
          name: `upscale-confirm-banner::${currentPlanCode}_to_${planCode}::success`,
          type: 'navigation',
        });
      })
      .catch((error) => {
        this.atInternet.trackPage({
          name: `upscale-confirm-banner::${currentPlanCode}_to_${planCode}::error`,
          type: 'navigation',
        });

Comment (anooparveti) [reply packages/manager/modules/vps/src/upscale/upscale.controller.js:495]: But you dont need to pass it to `performUpscale` function.

**Review by anooparveti — DISMISSED** [review]

File: packages/manager/modules/vps/src/upscale/upscale.controller.js [hunk 5]
```diff
@@ -411,16 +411,26 @@ export default class UpscaleController {
...
             eliteUpgrade: 'vps_upscale_get_configuration_information_error',
             normalUpgrade: 'vps_upscale_get_information_error',
           },
-        ),
-      )
+          this.atInternet.trackPage({
+            name: `upscale-confirm-banner::${planCode}::error`,
+            type: 'navigation',
+          }),
+        );
```
Comment (mohammed-zahaf) [inline packages/manager/modules/vps/src/upscale/upscale.controller.js:432]: tracking first then the rest.
put tracking between line 420 and 421.

**Review by anooparveti — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/modules/vps/src/dashboard/tile/configuration/upgrade/routing.js [hunk 6]
```diff
@@ -239,7 +238,17 @@ export default /* @ngInject */ ($stateProvider) => {
...
+            atInternet.trackPage({
+              name: `vps::upgrade-confirm-banner::success::${upgradeType}-${from}-to-${to}`,
+              type: 'navigation',
+            });
+          })
           .catch((error) => {
+            atInternet.trackPage({
+              name: `vps::upgrade-confirm-banner::error::${upgradeType}-${from}-to-${to}`,
+              type: 'navigation',
+            });
```
Comment (ghyenne) [inline packages/manager/modules/vps/src/dashboard/tile/configuration/upgrade/routing.js:251]: In packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js you have a function trackPage why you don't reuse it here ?

**Review by ghyenne — DISMISSED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Comment by anooparveti** [comment]
Rebasing with `release/sprint-2250-2252` with resolving conflicts.

**Review by anooparveti — APPROVED** [review]

**Pull request merged**